libraries/AP_VideoTX/AP_Tramp.cpp

```diff
@@ -0,0 +1,487 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+   Code by Andy Piper, ported from betaflight vtx_tramp
+*/
+
+#include "AP_Tramp.h"
+#include <AP_Math/crc.h>
+#include <GCS_MAVLink/GCS.h>
+#include <AP_HAL/utility/sparse-endian.h>
+
+#if HAL_TRAMP_ENABLED
+
+#define AP_TRAMP_UART_BAUD            9600
+// request and response size is 16 bytes
+#define AP_TRAMP_UART_BUFSIZE_RX      32
+#define AP_TRAMP_UART_BUFSIZE_TX      32
+
+// Define periods between requests
+#define TRAMP_MIN_REQUEST_PERIOD_US (200 * 1000) // 200ms
+#define TRAMP_STATUS_REQUEST_PERIOD_US (1000 * 1000) // 1s
+
+//#define TRAMP_DEBUG
+#ifdef TRAMP_DEBUG
+# define debug(fmt, args...)	do { hal.console->printf("TRAMP: " fmt "\n", ##args); } while (0)
+#else
+# define debug(fmt, args...)	do {} while(0)
+#endif
+
+extern const AP_HAL::HAL &hal;
+
+AP_Tramp::AP_Tramp()
+{
+    singleton = this;
+}
+
+AP_Tramp *AP_Tramp::singleton;
+
+// Calculate tramp protocol checksum of provided buffer
+uint8_t AP_Tramp::checksum(uint8_t *buf)
```

Comment:
Perhaps this should go into our checksum library in AP_Math?

Reply:
I think it's tramp specific